src/sql/workbench/browser/modal/modal.ts

```diff
@@ -396,14 +395,19 @@ export abstract class Modal extends Disposable implements IThemable {
 	protected hide() {
 		this._modalShowingContext.get()!.pop();
 		this._bodyContainer.remove();
-		if (this._focusedElementBeforeOpen) {
-			this._focusedElementBeforeOpen.focus();
-		}
 		this._keydownListener.dispose();
 		this._resizeListener.dispose();
 		this._telemetryService.createActionEvent(TelemetryKeys.TelemetryView.Shell, TelemetryKeys.ModalDialogClosed)
 			.withAdditionalProperties({ name: this._name })
 			.send();
+		this.restoreKeyboardFocus();
+	}
+
+	private restoreKeyboardFocus() {
+		if (this._focusedElementBeforeOpen && this._focusedElementBeforeOpen.offsetParent) { // when the element is visible, we can try to set focus to it directly.
```

Comment:
Why are you checking offsetParent here?

Reply:
i guess i don't have do this check anymore since we don't have to do the workaround. in my latest commit i have removed the check.